### PR TITLE
frontend to vite

### LINKED ISSUE #3785: Migrate frontend build to vite
**Describe the task**
To speed up builds and reloading we can migrate to vite and away from crac: https://vitejs.dev/

**Acceptance Criteria**
- [x] Frontend builds and deploys with vite as the build tool
- [x] All tests pass (may require migrating from jest to vitest: https://vitest.dev/)
- [x] All code coverage from jest and cypress tests are reported to codecov still

**Additional context**
- You can take a look at a previous attempt here: https://github.com/bcgov/wps/tree/experiment/migrate-to-vite, it's missing the last AC above requiring code coverage.

### PR DESCRIPTION
**Pull request by jmetev1**
Things I did
1. changed to vite and vitest
1.  removed some unneeded stuff from integration workflow
2. moved to node alpine bc couldn’t make it work with corepack and new type= module nature of vite
3. renamed env variables for vite
4. add global browser fetch in tests which was happening before behind the scenes with CRA
5. not running       "src/features/auth/slices/authenticationSlice.test.ts" bc ES Modules cannot be stubbed so need to find other solution for sinon.

Outcomes
1. can’t use require anymore unless you name file .cjs or .cts. Everything is default import export now. 
closes #3785

# Test Links:
[Landing Page](https://wps-pr-3860-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3860-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3860-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-3860-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3860-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3860-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3860-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3860-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/.yarnrc.yml [hunk 1]
```diff
@@ -1 +1,2 @@
 nodeLinker: node-modules
+enableHardenedMode: false
```
Comment (conbrad) [inline web/.yarnrc.yml:2]: Why this change?

File: web/index.html [hunk 2]
```diff
@@ -2,10 +2,7 @@
 <html lang="en">
   <head>
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <script src="config.js"></script>
-    <script type="text/javascript">
-      window.env = config
-    </script>
+    <script>console.log('abandon all hope')</script>
```
Comment (conbrad) [inline web/index.html:5]:
```suggestion
```

🤣

File: web/index.html [hunk 3]
```diff
@@ -21,15 +18,7 @@
...
-      This HTML file is a template.
-      If you open it directly in the browser, you will see an empty page.
-
-      You can add webfonts, meta tags, or analytics to this file.
-      The build step will place the bundled scripts into the <body> tag.
-
-      To begin the development, run `npm start` or `yarn start`.
-      To create a production bundle, use `npm run build` or `yarn build`.
-    -->
+    <div id="i be trippin"></div>
```
Comment (conbrad) [inline web/index.html:21]:
```suggestion
```

File: web/index.html [hunk 4]
```diff
@@ -21,15 +18,7 @@
...
+    <script type="module" src="/src/index.tsx"></script>
```
Comment (conbrad) [inline web/index.html:22]: Is this a change required by vite?

File: web/package.json [hunk 5]
```diff
@@ -51,73 +53,55 @@
...
-    "react-dom": "^18.2.0",
+    "react": "^18.3.1",
+    "react-dom": "^18.3.1",
     "react-redux": "^8.0.0",
     "react-router-dom": "^6.22.3",
     "recharts": "^2.1.8",
-    "typescript": "^5.0.0"
+    "typescript": "^5.2.2",
+    "vitest": "^2.0.5",
+    "whatwg-fetch": "^3.6.20"
```
Comment (conbrad) [inline web/package.json:63]: What's this dependency used for?

File: web/package.json [hunk 6]
```diff
@@ -51,73 +53,55 @@
...
   "scripts": {
-    "start": "CI=true craco start",
-    "start:cypress": "export $(cat .env.cypress | xargs) && craco --max_old_space_size=4096 -r @cypress/instrument-cra start",
-    "build": "craco --max_old_space_size=2048 build",
-    "build:prod": "craco --max_old_space_size=2048 build && yarn sentry:sourcemaps",
-    "test": "craco test --transformIgnorePatterns \"node_modules/(?!ol)/\"",
+    "dev": "vite",
+    "preview": "vite preview",
+    "dev:cypress": "export $(cat .env.cypress | xargs) && yarn dev",
+    "build:prod": "tsc -b && vite build",
```
Comment (conbrad) [inline web/package.json:69]: I think we need to keep sentry source map generation and upload here?

File: web/package.json [hunk 7]
```diff
@@ -51,73 +53,55 @@
...
     "eslint": "^9.7.0",
     "eslint-plugin-react-hooks": "^4.6.2",
+    "eslint-plugin-react-refresh": "^0.4.7",
     "globals": "^15.8.0",
-    "react-scripts": "^5.0.1",
     "start-server-and-test": "^2.0.0",
-    "ts-sinon": "^2.0.2"
-  },
-  "jest": {
-    "collectCoverageFrom": [
```
Comment (conbrad) [inline web/package.json:114]: This is also probably why coverage dropped, does vitest not have a similar configuration option?

File: web/src/app/store.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import { configureStore, AnyAction, ThunkAction } from '@reduxjs/toolkit'
 import thunk, { ThunkMiddleware } from 'redux-thunk'
-import rootReducer, { RootState } from 'app/rootReducer'
+import rootReducer, { RootState } from 'app/rootReducer.ts'
```
Comment (conbrad) [inline web/src/app/store.ts:3]: Is this change required?

File: web/src/features/moreCast2/forecastDraft.test.ts [hunk 9]
```diff
@@ -3,21 +3,22 @@ import { DraftMorecast2Rows } from 'features/moreCast2/interfaces'
 import { buildValidActualRow, buildValidForecastRow } from 'features/moreCast2/rowFilters.test'
 import { DateTime } from 'luxon'
 import * as DateUtils from 'utils/date'
+import { vi } from 'vitest'
```
Comment (conbrad) [inline web/src/features/moreCast2/forecastDraft.test.ts:6]: Should we also import `expect`, `it`, etc also?

File: web/src/index.tsx [hunk 10]
```diff
@@ -1,12 +1,11 @@
 import React from 'react'
 import * as ReactDOMClient from 'react-dom/client'
 import { Provider } from 'react-redux'
-import App from 'app/App'
+import App from 'app/App.tsx'
```
Comment (conbrad) [inline web/src/index.tsx:4]: Is this change required?

File: web/src/types/window.d.ts
Comment (conbrad) [inline web/src/types/window.d.ts:1]: Should these be `VITE_` prefixes instead of `REACT_APP`?

File: web/src/utils/env.ts [hunk 11]
```diff
@@ -27,6 +27,7 @@ if (process.env.NODE_ENV === 'production') {
     KC_AUTH_URL: window.env.REACT_APP_KEYCLOAK_AUTH_URL,
     KC_REALM: window.env.REACT_APP_KEYCLOAK_REALM,
     KC_CLIENT: window.env.REACT_APP_KEYCLOAK_CLIENT,
+    WF1_AUTH_URL: window.env.REACT_APP_WF1_AUTH_URL,
```
Comment (conbrad) [inline web/src/utils/env.ts:30]:
```suggestion
```
I don't think we need `WF1_AUTH_URL` anymore

File: web/tsconfig.node.json [hunk 12]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
+    "composite": true,
+    "tsBuildInfoFile": "./node_modules/.tmp/tsconfig.node.tsbuildinfo",
```
Comment (conbrad) [inline web/tsconfig.node.json:4]: Is this option required? Seems to reference a temporary file

File: web/vite.config.ts [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+      },
+    }),
+    , svgr(),
+    istanbul({
+      include: 'src/*',
+      exclude: ['node_modules', 'test/'],
+      extension: ['.js', '.ts'],
+      cypress: true
+    }),
+    sentryVitePlugin({
```
Comment (conbrad) [inline web/vite.config.ts:46]: Does this generated and upload source maps for us?

File: web/vite.config.ts [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+      authToken: process.env.SENTRY_AUTH_TOKEN,
+      sourcemaps: {
+        filesToDeleteAfterUpload: ['**/*.map'],
+      },
+    }),
+  ],
+  server: {
+    port: Number(process.env.PORT) || 3000,
+    watch: {
+      ignored: [
```
Comment (conbrad) [inline web/vite.config.ts:58]: Should we also ignore `*.test.*` files?

**Review by conbrad — COMMENTED** [review]
Looking good! Mostly questions. Would you mind also updating the PR description to include the summary of the changes?

Top level question, is it required that we move ts config files to common js files?

File: Dockerfile.web [hunk 15]
```diff
@@ -9,10 +9,10 @@ ARG NODE_OPTIONS="--v8-pool-size=4"
 # Pull from local registry - we can't pull from docker due to limits.
 # see https://catalog.redhat.com/software/containers/ubi9/nodejs-20/64770ac7a835530172eee6a9 for
 # details
-FROM registry.access.redhat.com/ubi9/nodejs-20 as static
-
+FROM node:20-alpine AS static
```
Comment (conbrad) [inline Dockerfile.web:12]: Is this required? Will the web still build with the old base image from redhat?

File: web/.yarnrc.yml (hunk 1 above)
Comment (jmetev1) [reply web/.yarnrc.yml:2]: temp thing for to make ci run faster. taking out now.

File: web/index.html (hunk 4 above)
Comment (jmetev1) [reply web/index.html:22]: yep. it's ok?

Comment (conbrad) [reply web/index.html:22]: Yeah just curious if you have a link to the docs about it or some background info.

Comment (conbrad) [reply web/index.html:22]: I think so, just wondering if there's any background info on why

File: web/src/features/moreCast2/forecastDraft.test.ts (hunk 9 above)
Comment (jmetev1) [reply web/src/features/moreCast2/forecastDraft.test.ts:6]: i agree this looks a little funky. but expect and it are coming from setuptests.ts already

File: web/package.json (hunk 6 above)
Comment (conbrad) [reply web/package.json:69]: Unless there's a vite plugin doing it now.

File: web/src/features/moreCast2/forecastDraft.test.ts (hunk 9 above)
Comment (conbrad) [reply web/src/features/moreCast2/forecastDraft.test.ts:6]: I see, why do other tests require imports?

File: Dockerfile.web (hunk 15 above)
Comment (jmetev1) [reply Dockerfile.web:12]: Nope. There's just something about the redhat node images. I've tried a lot of things. The error is "Dynamic require of "util" is not supported". And googling it is not very helpful. So we can't use docker bc we don't have a paid account and since it's not cached we will exceed rate limit? It seems like somehow yarn/corepack is part of the problem. Want to switch to npm instead of yarn? I don't have a lot more ideas.

File: web/index.html (hunk 4 above)
Comment (jmetev1) [reply web/index.html:20]: the beginning of this is good intro. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules Vite is all about using modern native  esm support.

File: Dockerfile.web (hunk 15 above)
Comment (conbrad) [reply Dockerfile.web:12]: All good just wanted to clarify. Let's keep this and adjust if we run into rate limits.

File: web/vite.config.ts (hunk 14 above)
Comment (jmetev1) [reply web/vite.config.ts:58]: i added these for cypress tests running. When that is running these directories get changed and so dev server restarts. But the test files won't get changed so I think we're ok.

Comment (conbrad) [reply web/vite.config.ts:58]: Does this config only apply to cypress tests though? If a developer is running the vite server locally and making changes to component tests will this trigger dev server restarts?

**Comment by conbrad** [comment]
There's a bunch of sonarcloud-flagged unused imports from `vitest` in test files, e.g. https://sonarcloud.io/project/issues?pullRequest=3860&open=AZFsFi30uttMHqCURf1j&id=bcgov_wps

Would you mind going through those and removing unused imports?

File: .github/workflows/integration.yml [hunk 16]
```diff
@@ -191,8 +191,6 @@ jobs:
         with:
           # For sonar-scanner to work properly we can't use a shallow fetch.
           fetch-depth: 0
-      - name: Setup kernel for react, increase watchers
```
Comment (jmetev1) [inline .github/workflows/integration.yml:194]: i don't think this is needed.

File: .vscode/settings.json [hunk 17]
```diff
@@ -8,7 +8,6 @@
         ".github": false,
         "**/.pytest_cache": true,
         "**/__pycache__**": true,
-        "web/node_modules": true,
```
Comment (jmetev1) [inline .vscode/settings.json:11]: just a personal preference. I'm always going in node modules. But other people want it hidden?

File: web/.env.cypress [hunk 18]
```diff
@@ -1,9 +1,9 @@
 BROWSER=none
 PORT=3030
-REACT_APP_API_BASE_URL=http://localhost:8080/api
-REACT_APP_HIDE_DISCLAIMER=false
-REACT_APP_KEYCLOAK_AUTH_URL=https://keycloak-auth-url.ca
-REACT_APP_SM_LOGOUT_URL=https://sm-logout-url.ca
-REACT_APP_KEYCLOAK_REALM=8wl6x4cp
-REACT_APP_KEYCLOAK_CLIENT=wps-web
-REACT_APP_PMTILES_BUCKET=https://nrs.objectstore.gov.bc.ca/gpdqha/psu/pmtiles/
```
Comment (jmetev1) [inline web/.env.cypress:9]: i hadn't noticed this. so cypress is using the real bucket but not other tests

File: .vscode/settings.json (hunk 17 above)
Comment (conbrad) [reply .vscode/settings.json:11]: I think it's fine. Where it can kind of get annoying is if you're doing a global string search in the repo and node modules are indexed those results can cloud what you're looking for.

File: web/.env.cypress (hunk 18 above)
Comment (conbrad) [reply web/.env.cypress:9]: Other tests could use the real bucket (these are public, I forgot)

File: web/package.json (hunk 7 above)
Comment (jmetev1) [reply web/package.json:114]: yep it's in vitest.config test/coverage

File: web/src/features/moreCast2/forecastDraft.test.ts (hunk 9 above)
Comment (jmetev1) [reply web/src/features/moreCast2/forecastDraft.test.ts:6]: hmmmm. good question. will look.

File: web/src/types/window.d.ts
Comment (jmetev1) [reply web/src/types/window.d.ts:1]: these are the ones that are coming from the config file that is replaced when deployed. so we would have to rename them there and here. we can do that if you like that better

**Comment by jmetev1** [comment]
> Looking good! Mostly questions. Would you mind also updating the PR description to include the summary of the changes?
> 
> Top level question, is it required that we move ts config files to common js files?

Yep I can add more details.
If config or all other files use require then they have to either be rewritten or they have to be renamed to .cjs. Bc package.json now has type:module, all .js or .ts files are now assumed to be esm compliant. I rewrote some and others like cypress config I couldn't make work so just renamed to cjs.

**Comment by jmetev1** [comment]
My numbers are improving now that one less test is disabled!
image

File: .vscode/settings.json (hunk 17 above)
Comment (jmetev1) [reply .vscode/settings.json:11]: do you mean when you're searching using vscode? on my end at least if this button is clicked then it will ignore node_modules 
image

File: web/vite.config.ts (hunk 14 above)
Comment (jmetev1) [reply web/vite.config.ts:49]: yea. is that a problem? it's like instantatenous. but i can add tests to ignored.

File: web/vite.config.ts (hunk 13 above)
Comment (jmetev1) [reply web/vite.config.ts]: i went back to old way

File: web/tsconfig.node.json (hunk 12 above)
Comment (jmetev1) [reply web/tsconfig.node.json]: removed

File: web/src/app/store.ts (hunk 8 above)
Comment (jmetev1) [reply web/src/app/store.ts]: removed

File: web/package.json (hunk 5 above)
Comment (jmetev1) [reply web/package.json:66]: will add to summary notes

File: web/src/features/fba/components/map/fbaMap.test.tsx [hunk 19]
```diff
@@ -3,10 +3,16 @@ import store from 'app/store'
...
 import { RunType } from 'features/fba/pages/FireBehaviourAdvisoryPage'
 import { DateTime } from 'luxon'
-import React from 'react'
+
 import { Provider } from 'react-redux'
+// @ts-ignore
+import { fetch } from 'whatwg-fetch'
 
 describe('FBAMap', () => {
+  beforeEach(() => {
```
Comment (conbrad) [inline web/src/features/fba/components/map/fbaMap.test.tsx:12]: Why do we need to overwrite the global fetch function?

Comment (jmetev1) [reply web/src/features/fba/components/map/fbaMap.test.tsx:12]: in create react app version this was also being done, just under the hood. without this, when olpmtiles does a fetch to that s3 bucket it will expect browser fetch but will actually get node built in fetch and error out. this was a fascinating discovery that took me so many hours to figure out. other options would be to mock some sub components i suppose. or the whole fetch response.

Comment (conbrad) [reply web/src/features/fba/components/map/fbaMap.test.tsx:12]: Interesting... would it make sense to replace the node fetch in an `after` function? Or perhaps just set this as the global fetch for all tests in `setupTests`?

File: .vscode/settings.json (hunk 17 above)
Comment (conbrad) [reply .vscode/settings.json:11]: Yeah in vscode, interesting ok, will check that button out.

File: web/vite.config.ts (hunk 14 above)
Comment (conbrad) [reply web/vite.config.ts:49]: Not sure if it's a problem, we could add that after if it becomes one.

File: web/src/types/window.d.ts
Comment (conbrad) [reply web/src/types/window.d.ts:1]: Actually if we could remove this `window.d.ts` file altogether that'd be nice, one less place to add env vars.

File: web/src/features/fba/components/map/fbaMap.test.tsx (hunk 19 above)
Comment (jmetev1) [reply web/src/features/fba/components/map/fbaMap.test.tsx]: sure thing. done

File: web/.env.cypress (hunk 18 above)
Comment (jmetev1) [reply web/.env.cypress:9]: done

File: web/src/utils/env.ts (hunk 11 above)
Comment (jmetev1) [reply web/src/utils/env.ts]: done

**Comment by conbrad** [comment]
> My numbers are improving now that one less test is disabled! image

Nice! How many tests are disabled? The code coverage reports make sense when I'm looking at the ones with coverage or partial coverage, but a bunch of missing coverage around HFI Calculator signals that either those cypress tests are disabled or the coverage report isn't being merged properly.

**Comment by jmetev1** [comment]
> > My numbers are improving now that one less test is disabled! image
> 
> Nice! How many tests are disabled? The code coverage reports make sense when I'm looking at the ones with coverage or partial coverage, but a bunch of missing coverage around HFI Calculator signals that either those cypress tests are disabled or the coverage report isn't being merged properly.

image
whoops!

File: web/src/whatwg-fetch.d.ts [hunk 20]
```diff
@@ -0,0 +1 @@
+declare module "whatwg-fetch";
```
Comment (jmetev1) [inline web/src/whatwg-fetch.d.ts:1]: sometimes typescript just wears me out. i don't even know what this does. i just want to polyfill fetch in the test. other suggestions accepted.

Comment (conbrad) [reply web/src/whatwg-fetch.d.ts:1]: I imagine `whatwg-fetch` doesn't ship with type definitions, I think this is fine since we're just using it for tests. What errors were you seeing?

Comment (conbrad) [reply web/src/whatwg-fetch.d.ts:1]: https://www.npmjs.com/package/@types/whatwg-fetch says "fetch types are now provided by '--lib dom'"

File: web/src/types/window.d.ts
Comment (jmetev1) [reply web/src/types/window.d.ts:1]: like right now? than typescript errors out. i do really want to fix the whole env situation but i'm trying not to avoid mission creep.

Comment (conbrad) [reply web/src/types/window.d.ts:1]: Ok yeah no problem.

File: web/package.json [hunk 21]
```diff
@@ -34,13 +35,15 @@
     "@types/webpack-env": "^1.15.1",
     "@typescript-eslint/eslint-plugin": "^8.0.0",
     "@typescript-eslint/parser": "^8.0.0",
+    "@vitest/coverage-istanbul": "^2.0.5",
```
Comment (conbrad) [inline web/package.json:38]: Nit: can this go in `devDependencies`?

File: web/vite.config.ts [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+import svgr from 'vite-plugin-svgr'
+import istanbul from 'vite-plugin-istanbul'
+
+export default defineConfig({
+  build: {
+    sourcemap: true,
+    outDir: 'build'
+  },
+  optimizeDeps: {
+    include: ['@mui/material/Tooltip', '@emotion/styled/base', '@mui/material/Unstable_Grid2']
```
Comment (conbrad) [inline web/vite.config.ts:14]: Were these added to avoid the vite build errors you mentioned earlier?

Comment (jmetev1) [reply web/vite.config.ts:14]: yep

Comment (conbrad) [reply web/vite.config.ts:14]: Ok cool, is it possible to add a comment in this config ts file explaining that a little bit with links to the issue?

File: web/package.json (hunk 21 above)
Comment (jmetev1) [reply web/package.json:38]: indulge a digression. I have this conviction that there is no distinction between dependencies and dev dependencies in a front end non library project. like the whole thing gets built. there are no real dependencies after that. right? also by any measure i feel like most of the classification here is wrong. should typescript, and sentry cli and all the types also be moved to dev deps?

Comment (conbrad) [reply web/package.json:38]: Digression indulged. Yes seems like `devDependencies` also get installed since we don't install in the web image with the `--production` flag. Perhaps we should add that but can be deferred in a follow up PR.

**Review by conbrad — APPROVED** [review]
Looks great, amazing work thanks for the considerable amount of effort that went into this! Builds are faster and we can sever our dependency on `cra`. 

On my side just the couple comments I added this morning to be resolved, and the injected error to test sentry source maps and I'm happy. I'll let @dgboss and @brettedw carve out some time to review further.

File: web/src/app/store.ts [hunk 23]
```diff
@@ -14,16 +14,6 @@ const store = configureStore({
     getDefaultMiddleware({ immutableCheck: false, serializableCheck: false }).concat(thunkMiddleware)
 })
 
-if (process.env.NODE_ENV === 'development' && module.hot) {
```
Comment (dgboss) [inline web/src/app/store.ts:17]: Do you know if we get the equivalent behaviour with vite out of the box? Or will we have to re-start our debugging session if we are making changes to a reducer?

File: web/src/features/landingPage/components/Sidebar.tsx [hunk 24]
```diff
@@ -15,8 +16,8 @@ import Typography from '@mui/material/Typography'
 import useMediaQuery from '@mui/material/useMediaQuery'
 import SidebarToolList from 'features/landingPage/components/SidebarToolList'
 import Subheading from 'features/landingPage/components/Subheading'
-import { ReactComponent as MsTeamsIcon } from 'features/landingPage/images/msTeams.svg'
-import { ReactComponent as MiroIcon } from 'features/landingPage/images/miro.svg'
+import MsTeamsIcon from 'features/landingPage/images/msTeams.svg?react'
```
Comment (dgboss) [inline web/src/features/landingPage/components/Sidebar.tsx:19]: Interesting syntax! I haven't seen this before. Nice!

**Review by dgboss — APPROVED** [review]
Amazing work! Thank you for all the time you put into this!

File: web/src/app/store.ts (hunk 23 above)
Comment (jmetev1) [reply web/src/app/store.ts:17]: hmmm. I feel like vite should handle this well since a big focus of it is not having to rebundle stuff https://vitejs.dev/guide/why.html#why-vite . But i haven't done much reducer stuff lately. Can you break down for me what problem this was solving? you don't want to lose some state in redux?

File: web/src/features/landingPage/components/Sidebar.tsx (hunk 24 above)
Comment (jmetev1) [reply web/src/features/landingPage/components/Sidebar.tsx:19]: it's a little freaky eh

File: web/package.json (hunk 21 above)
Comment (jmetev1) [reply web/package.json:38]: ahh but that wouldn't work. to do the build we need most of the stuff that is in dev dependencies. i think it's a distinction that makes sense in backend projects and just got carried into frontend somehow.

Comment (jmetev1) [reply web/package.json:38]: but the multistage build means node modules get's left behind anyway so all good.

File: web/vite.config.ts (hunk 22 above)
Comment (jmetev1) [reply web/vite.config.ts:16]: done

File: web/src/whatwg-fetch.d.ts (hunk 20 above)
Comment (jmetev1) [reply web/src/whatwg-fetch.d.ts:1]: that seems to mean i need tsconfig compileroptions: [dom]. but i already had that and still got warnings.

File: web/src/app/store.ts (hunk 23 above)
Comment (dgboss) [reply web/src/app/store.ts:17]: I was curious more than anything so not a big deal. I think you are correct in that it prevented loss of state in redux. I don't think we need to do anything or look into this any further. Thanks!

File: web/src/whatwg-fetch.d.ts (hunk 20 above)
Comment (conbrad) [reply web/src/whatwg-fetch.d.ts:1]: Ok np

**Review by brettedw — APPROVED** [review]
Thank you for the effort you put into this @jmetev1, love it!
